packages/markdown/src/lib/remark-slate/remarkPlugin.ts

```diff
@@ -1,16 +1,21 @@
 /* eslint-disable @typescript-eslint/ban-ts-comment */
 
+import type { Processor } from 'unified';
+
 import type { MdastNode, RemarkPluginOptions } from './types';
 
-import { remarkTransformNode } from './remarkTransformNode';
+import { MarkdownPlugin } from '../MarkdownPlugin';
+import { remarkDefaultCompiler } from './remarkDefaultCompiler';
+import { remarkSplitLineBreaksCompiler } from './remarkSplitLineBreaksCompiler';
 
-export function remarkPlugin(options: RemarkPluginOptions) {
-  const compiler = (node: { children: MdastNode[] }) => {
-    return node.children.flatMap((child) =>
-      remarkTransformNode(child, options)
-    );
-  };
+export function remarkPlugin(
+  this: Processor<undefined, undefined, undefined, MdastNode, any>,
+  options: RemarkPluginOptions
+) {
+  const shouldSplitLineBreaks =
+    options.editor.getOptions(MarkdownPlugin).splitLineBreaks;
 
-  // @ts-ignore
-  this.Compiler = compiler;
+  this.compiler = shouldSplitLineBreaks
+    ? (tree: MdastNode) => remarkSplitLineBreaksCompiler(tree, options)
+    : (tree: MdastNode) => remarkDefaultCompiler(tree, options);
```

Comment:
We still appear to be ignoring the `options` variable if `splitLineBreaks` is enabled. Is there any way of removing this conditional and using a single compiler that consumes `splitLineBreaks` as an option?

Reply:
Oh! Thank you so much